pyrobosim/pyrobosim/core/world.py

```diff
@@ -683,13 +687,18 @@ def check_occupancy(self, pose):
         :return: True if the pose is occupied, else False if free.
         :rtype: bool
         """
+        if self.occupancy_grid:
+            x, y = (pose.x, pose.y) if isinstance(pose, Pose) else (pose[0], pose[1])
+            x, y = self.occupancy_grid.xy_from_world(x, y)
+            return self.occupancy_grid.data[x, y] == 1
         # Loop through all the rooms and hallways and check if the pose
         # is deemed collision-free in any of them.
-        for entity in itertools.chain(self.rooms, self.hallways):
-            if entity.is_collision_free(pose):
-                return False
-        # If we made it through, the pose is occupied.
-        return True
+        else:
```

Comment:
Maybe add a comment here, or in the function description saying that if there is an occupancy grid, we should use that to check, else we go through all the polygons and check that way.